### PR TITLE
Community DB-API & API with their Tests

### PR DESCRIPTION
**Pull request by guillecro**
Hi! I come today to give you the Community API and DB-API working properly.
Also their tests were implemented and the API was documented.

To build the API run `npm run docs:build` and you will have it available on `http://localhost:3000/docs/api`

### CODE REVIEW

**Comment by guillecro** [comment]
Note: If you download the repo and try the tests or the dev env, it wont work.
Here there are no default values. You have to make a `.env` file according to the one provided and set up keycloak with their realms for dev env and test env.

On the root folder of the project there are available a few realms for prod, dev and test that you can use and import inside your keycloak instance.

File: server.js [hunk 1]
```diff
@@ -15,10 +15,13 @@ const loggerMiddleware = expressWinston.logger({ winstonInstance: log })
...
   try {
-    console.log('================================================')
-    console.log('Starting server with the following configuration')
-    console.log(config)
-    console.log('================================================')
+    if (NODE_ENV != 'test') {
+      console.log('================================================')
+      console.log(`Starting server [${NODE_ENV}] with the following config`)
+      console.log('================================================')
+      console.log(config)
```
Comment (franciclo) [inline server.js]: i don't is a good idea to log the whole config..

**Comment by franciclo** [comment]
im not able to start the app due to some error in the docker-compose related to mysql volume

**Comment by guillecro** [comment]
The docker compose needs to be reviewed in another issue. Sadly my
knowledge on docker are not that great for this task.

About logging the config, it could be useful on Dev environment, as many
backend framework does (like next.js)

File: server.js (hunk 1 above)
Comment (franciclo) [reply server.js]: sorry didn't see the conditional

**Comment by franciclo** [comment]
Yes, i missed that conditional about the logging.
How are you testing it without the docker-compose?

**Comment by guillecro** [comment]
I downloaded jboss/keycloak docker and i just started it as independent docker-compose from the one we have in this repo.

I will make a new issue stating that I have to deal with that configuration. Might be the next step, so it can be more clear for you. I need to make some research about docker thou, like, how to run scripts and manage enviroment variables to initialize the keycloak configuration and its users with the CLI commands.

File: server.js (hunk 1 above)
Comment (guillecro) [reply server.js]: Changed it so it is only available for dev enviroment!

**Pull request merged**